src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,273 @@
+package edu.harvard.iq.dataverse.util;
+
+/**
+ * Eko Indarto, DANS
+ * Vic Ding, DANS
+ *
+ * This file prepares the resources used in Signposting
+ *
+ * It requires correspondence configuration to function well.
+ * The configuration key used is SignpostingConf.
+ * It is a json structure shown below
+ *
+ * useDefaultFileType is an on/off switch during linkset creating time, it controls whether the default type is
+ * used, which is always Dataset
+ * {
+ *   "indetifier-schema": {"ORCID":"https://orcid.org/", "ISNI":"https://isni.org/isni/", "ScopusID":"https://www.scopus.com/authid/detail.uri?authorId="},
+ *   "license": {"CC0":"https://creativecommons.org/licenses/by/4.0/", "MIT": "https://url", "APACHE":"https://url"},
+ *   "cite-as": {"doi":"https://citation.crosscite.org/format?style=bibtex&doi=", "type":"application/vnd.datacite.datacite+json"},
+ *   "useDefaultFileType": true,
+ *   "defaultFileTypeValue": "https://schema.org/Dataset"
+ * }
+ *
+ * The configuration can be modified during run time by the administrator.
+ *
+ */
+
+import edu.harvard.iq.dataverse.*;
+import edu.harvard.iq.dataverse.dataaccess.StorageIO;
+import edu.harvard.iq.dataverse.dataaccess.SwiftAccessIO;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.util.json.JsonPrinter;
+
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.StringReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.json.NullSafeJsonBuilder.jsonObjectBuilder;
+
+public class SignpostingResources {
+    private static final Logger logger = Logger.getLogger(SignpostingResources.class.getCanonicalName());
+    SystemConfig systemConfig;
+    DatasetVersion workingDatasetVersion;
+    JsonObject idschemaJsonObj;
+    JsonObject licJsonObj;
+    JsonObject citeAsJsonObj;
+    Boolean useDefaultFileType;
+    String defaultFileTypeValue;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting){
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
```

Comment:
This long string is very hard to read, because of all the escaping that has to be done. Could you load it from an embedded .json file?